ui/scripts/sharedFunctions.js

```diff
@@ -2851,3 +2851,34 @@ cloudStack.createArrayOfParametersForCreatePodCommand = function (zoneId, data){
     cloudStack.addParameterToCommandUrlParameterArrayIfValueIsNotEmpty(array, "endIp", data.podEndIp);
     return array;
 }
+
+cloudStack.listDiskOfferings = function(options){
+    var defaultOptions = {
+            listAll: false,
+            isRecursive: false,
+            error: function(data) {
+                args.response.error(data);
+            }
+    };
+    var mergedOptions = $.extend({}, defaultOptions, options);
+    
+    var listDiskOfferingsUrl = "listDiskOfferings";
+    if(mergedOptions.listAll){
+        listDiskOfferingsUrl = listDiskOfferingsUrl + "&listall=true";
+    }
+    if(mergedOptions.isRecursive){
+        listDiskOfferingsUrl = listDiskOfferingsUrl + "&isrecursive=true";
+    }
+    var diskOfferings = undefined;
+    $.ajax({
+        url: createURL(listDiskOfferingsUrl),
+        data: mergedOptions.data,
```

Comment:
For some calls, we don't pass 'data' in the parameter list. For e.g. [var diskOfferings = cloudStack.listDiskOfferings({listAll: true});] 

I think, it would be better if we check the existance of the parameter before assigning.

Reply:
I am not checking the existence of `mergedOptions.data` before ahe ssignment because the result would be the same.  I mean, if `mergedOptions.data` is `undefined`, than the object `data` was not provided. Therefore, the object `data` in the `ajax` function will have the value `undefined`. If I add an `IF` condition and decide to assign `data`  based on the existence of `mergedOptions.data`; then, when `mergedOptions.data == undefined` (I would not assign it), the `data` object of `ajax` would also be undefined. The result is the same, and thus there is no reason to add extra code.

Was my explanation clear?

Reply:
@rafaelweingartner, Got it. Thanks.